code/modules/paperwork/paperplane.dm

```diff
@@ -69,13 +60,12 @@
 		for(var/S in stamped)
 			. += "paperplane_[S]"
 
-/obj/item/paperplane/attack_self(mob/user)
+/obj/item/paperplane/attack_self(mob/user) //[/obj/item/paperplane/Exited()] will delete src as the internal paper is moved out.
 	to_chat(user, "<span class='notice'>You unfold [src].</span>")
-	var/obj/item/paper/internal_paper_tmp = internalPaper
-	internal_paper_tmp.forceMove(loc)
-	internalPaper = null
-	qdel(src)
-	user.put_in_hands(internal_paper_tmp)
+	if(user.Adjacent(internalPaper))
+		user.put_in_hands(internalPaper)
+	else
+		internalPaper.forceMove(loc)
```

Comment:
I think clarity is better then a comment, could you explain your thoughts on this?